hack/implementations/envoy-gateway/create-cluster.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
```

Comment:
I'm torn on this script. It feels like it should either exist in Envoy Gateway or in a more generic place within GW API. If it were within GW API, it seems like it could just be a generic script for creating a kind cluster with metal LB for the sake of testing with ~any in-cluster implementation of Gateway API.

Reply:
ah would you prefer a `curl` / `wget` that fetches this script and runs it ?

yah its a generic script which can be used for any implementation - should i move it under `hack/implementations/common` ?

Reply:
I think that a `curl` or `wget` that fetches this script from EG is probably better for now.

It might be good to have something similar in the Gateway API repo, but that's probably a larger discussion that we shouldn't block the rest of this PR for.

Reply:
We already have a place where such a piece of documentation should be put: https://github.com/kubernetes-sigs/gateway-api/tree/main/conformance/reports/v1.1.0/envoy-gateway. I guess that running `CONFORMANCE_REPORT_PATH=conformance-report-k8s.yaml make experimental-conformance` reproduces the same result as here (and if not, I think we should update the reports README instead of putting this script along with README here). I lean toward leaving this kind of script out of the Gateway API repo

Reply:
over the past years, Ive seen many contributors find it hard to test their PRs associated with adding/modifying conformance tests, This was an attempt to make it easier for a contributor to say `Hey I tested this Conformance Test PR on these X platforms by following the steps in ...`,  thereby making it easier for new contributors to come in and pick up conformance issues, contributions seems to be on the decline rn. 

The steps in the conformance tests are not sufficient.

Happy to close this PR if the maintainers prefer that

Reply:
I also lean towards leaving this kind of script out of the repository. README's for our reporting implementations seem entirely fine (and quite helpful for users), but I don't want to set a precedent for implementation-specific executable code as something that we host here in the upstream (and if that precedent has already been set somehow, we should re-look at that).

Reply:
> but I don't want to set a precedent for implementation-specific executable code as something that we host here in the upstream

Mostly agree with this reasoning, but I think this specific script would be useful for ~any implementation that could be deployed inside a cluster. I think I'd be fine with either bumping this script up to a more generic place that could be used to test with any in-cluster implementation, or just updating the directions to describe how to run conformance tests with Envoy Gateway.

I'd also agree that we shouldn't open this broader idea to just any implementation, maybe we can limit it to implementations that have a solid track record of delivering passing conformance reports?

Reply:
I can be persuaded, putting it in a general place and so forth would certainly be preferable.

Reply:
ive moved the `create-cluster.sh` to `hack/implementations/common` so any implementation can peruse it